discovery_engine_core/core/src/engine.rs

```diff
@@ -283,15 +289,27 @@ where
         let mut errors = Vec::new();
         for stack in self.stacks.write().await.values_mut() {
             if stack.len() <= request_new {
-                match stack.ops.new_items(key_phrases, &self.ranker).await {
+                let articles = stack.ops.new_items(key_phrases).await;
+                match articles.map_err(Error::StackOpGen).and_then(|articles| {
+                    let id = stack.id();
+                    articles
```

Comment:
We want to run `filter_articles` on them before running smbert

Reply:
ok. I've combined "`new_items` followed by `filter_articles`" into a single `Stack` method `filter_new_articles`, which is probably where it ought to belong. so now in `update_stacks`:
```rust
                let articles = stack.filter_new_articles(key_phrases).await;
                match articles.map_err(Error::StackOpFailed).and_then(|articles| {
                    // the rest same as before ...
```
which also eliminates the `Error::StackOpGen` variant I introduced so overall it's a bit tidier (see the other comments)

Reply:
But they are two logical different things so I'm more to keep them separate, but we can keep the new error variants and use them to aovid `StackOpGen`.